design/object-store-realm.md

```diff
@@ -0,0 +1,214 @@
+3. Any non-master Ceph zonegroup is created in the realm if it does not exist already
+4. A master zone is created for the zonegroup.
+
+After these steps the admin should start up object stores configured with the same zone, zonegroup, and realm as configured (see next section).
+
+#### Notes
+1. The zonegroups section of the object-store-realm CR is only additive. If the playbook is reapplied and zonegroup already exist the zonegroup won't be recreated. If a zonegroup previously created is removed from the list of zonegroups, it will not be deleted. If a new zonegroup is added to the list it will be created. 
+2. The same principle applies to master-zones.
+3. When resource is deleted, realm and zonegroups are not deleted from the cluster. Realm and zonegroup deletion must be done through toolboxes.
+
+### Creating a Zone on the same cluster the object-store-realm was created.
+
+Once the storage admin creates a realm and zonegroup(s) on a cluster, the admin is ready to add object-stores to a master zone on a realm or add another zone to the zonegroup. 
+The admin will specify his/her configuration settings in the realm section of the object-store like in the following `object-store.yaml`.
+
+All objectstores with the realm section must be named in the form `realm_name` + `zone_name`.`zone_name` can be the name of a master zone created in the object store realm CR or the name of any other one..
+
+If the name of the objectstore is `realm_name` + '-' + `master_zone_name`, the the object stores created will be added to the master zone.
+
+If the name of the objectstore is not `realm_name` + '-' + `master_zone_name`, then a new zone will be created with the object stores created via the resource.
+
+This example config will add objectstores to the already created zone `zone-y`.
+These objectstores will go into the master zone, zonegroup, and realm created for the ObjectStoreRealm resource above.
+```yaml
+apiVersion: ceph.rook.io/v1alpha1
+kind: CephObjectStore
+metadata:
+  name: my-realm-zone-y
+  namespace: rook-ceph
+spec:
+  metadataPool:
+    failureDomain: host
+    replicated:
+      size: 3
+  dataPool:
+    failureDomain: device
+    erasureCoded:
+      dataChunks: 6
+      codingChunks: 2
+  gateway:
+    type: s3
+    port: 80
+    securePort: 443
+    instances: 3
+  realm:
+    name: my-realm
+    zonegroup: group-a
+```
+
+Now create the object store.
+```bash
+kubectl create -f object-store.yaml
+```
+#### Steps
+
+At this point the Rook operator recognizes that a new multisite object store resource needs to be configured. The operator will create all of the resources to start the object store.
+
+1. If the name of the object store is not the name of a master zone a Ceph zone is created as long as the zonegroup and realm have been created.
+2. If the name of the resource has a master zone name in it, the objectstores are added to the zone
+3. The Ceph mulitsite period is commited.
+
+#### Notes
```

Comment:
How about a section on deletion? For example, are we ever going to delete realms and zonegroups when CRs are deleted?